src/xb-builder.c

```diff
@@ -657,7 +656,6 @@ xb_builder_import_node(XbBuilder *self, XbBuilderNode *bn)
 {
 	XbBuilderPrivate *priv = GET_PRIVATE(self);
 	g_return_if_fail(XB_IS_BUILDER(self));
-	g_return_if_fail(XB_IS_BUILDER_NODE(bn));
```

Comment:
You could keep this precondition check in, but change it to `bn != NULL` (one of the implications of `XB_IS_BUILDER_NODE(bn)` is that `bn` must be non-`NULL`)